### PR TITLE
DRAFT: MOTD action

### LINKED ISSUE #42: Add MOTD action
Add 'MOTD' intent and action for sending user greeting at start of conversation. The relevant greeting - based on the date, e.g. Christmas, Easter - should be extracted from a .yml file via a custom action (or depending on the .yml file structure, can possibly just be listed as a normal response in a separate file).

### PR DESCRIPTION
**Pull request by bjarkiarmanns**
Closes #42

### CODE REVIEW

File: domain.yml [hunk 1]
```diff
@@ -191,6 +192,8 @@ responses:
   - text: Hvað verður um ís sem er búinn að vera lengi í boxi? Hann kemst á endanum í form.
   utter_operator:
   - text: Ég get því miður ekki tengt þig við þjónustufulltrúa. Opnunartímar þjónustuvers eru mánudaga til föstudaga frá kl. 09:00 til 15:00. Sími {phone}, netfang {email}
+  utter_operator_no_info:
+  - text: Ég get því miður ekki tengt þig við þjónustufulltrúa.
```
Comment (lumpidu) [inline domain.yml:196]: this slipped into the PR, could you create a branch directly from main ?

File: actions/actions.py [hunk 2]
```diff
@@ -277,8 +302,14 @@ def run(
...
-        r = res[0]
-        dispatcher.utter_message(response="utter_operator", phone=r.phone, email=r.email)
+        if res:
+            r = res[0]
+            if r.phone and r.email:
+                dispatcher.utter_message(response="utter_operator", phone=r.phone, email=r.email)
+            else:
+                dispatcher.utter_message(response="utter_operator_no_info")
+        else:
+            dispatcher.utter_message(response="utter_operator_no_info")
```
Comment (lumpidu) [inline actions/actions.py]: here too

File: tests/test_actions.py [hunk 3]
```diff
@@ -79,3 +79,21 @@ def test_get_info_for_contact(
...
+def test_get_motd(
+    tracker: Tracker,
+    dispatcher: CollectingDispatcher,
+    domain: DomainDict,
+    expected_events: Any
+):
+    action = actions.ActionGetMOTD()
+    actual_events = action.run(dispatcher=dispatcher, tracker=tracker, domain=domain)
+
+    assert actual_events == expected_events
```
Comment (lumpidu) [inline tests/test_actions.py:99]: I am missing the yaml file inside `data/motd/motd.yml`. Also the tests should be extended for edge cases, e.g. if a data starts and ends at the same day, and also some tests for if the default is used in case the date is different. You can mock the current date for your tests like mentioned here via "freezegun" https://stackoverflow.com/questions/23988853/how-to-mock-set-system-date-in-pytest

**Comment by lumpidu** [comment]
Could you also add an explanation for the MOTD action to the `Readme.md` ? I think it's quite relevant for the user to know, how this should be implemented.
